torch/csrc/LinalgBackend.cpp

```diff
@@ -0,0 +1,81 @@
+#include <torch/csrc/LinalgBackend.h>
+
+#include <ATen/LinalgBackend.h>
+
+#include <torch/csrc/Exceptions.h>
+#include <torch/csrc/utils/object_ptr.h>
+#include <torch/csrc/utils/python_strings.h>
+
+#include <structmember.h>
+#include <cstring>
+#include <string>
+
+PyObject *THPLinalgBackend_New(at::LinalgBackend linalg_backend)
+{
+  const std::string py_repr = at::LinalgBackendToRepr(linalg_backend);
+  auto type = (PyTypeObject*)&THPLinalgBackendType;
+  auto self = THPObjectPtr{type->tp_alloc(type, 0)};
+  if (!self) throw python_error();
+  auto self_ = reinterpret_cast<THPLinalgBackend*>(self.get());
+  self_->linalg_backend = linalg_backend;
+  std::strncpy (self_->name, py_repr.c_str(), LINALG_BACKEND_NAME_LEN);
+  self_->name[LINALG_BACKEND_NAME_LEN] = '\0';
+  return self.release();
+}
+
+PyObject *THPLinalgBackend_repr(THPLinalgBackend *self)
+{
+  return THPUtils_packString(self->name);
+}
+
+PyTypeObject THPLinalgBackendType = {
+  PyVarObject_HEAD_INIT(nullptr, 0)
+  "torch.linalg_backend",                       /* tp_name */
```

Comment:
Why do you need this full fledge custom objects?
If you just want to bind your union to python, you can do https://github.com/pytorch/pytorch/blob/dbbb02474b9ad8044d061caee9fb0916da93101b/torch/csrc/Module.cpp#L970-L990

Reply:
Thanks! Didn't know this pybind usage. I'll try to use this to simplify the PR.